test/ui-test/testSuites/suite_statusLogin/shared/steps/steps.py

```diff
@@ -39,23 +45,43 @@ def step(context):
     # Check valid process behavior:
     loginProcess.execute_process(True)
 
+
 @When("the user tries to login with invalid credentials")
 def step(context):
     loginProcess = context.userData['process']
     
     # Check invalid process behavior:
     loginProcess.execute_process(False)
 
+
 @Then("the user is able to login to Status Desktop application")
 def step(context):
     get_process_result(context)
+
     
 @Then("the user is NOT able to login to Status Desktop application")
 def step(context):
     get_process_result(context)
-  
+    
+
+@Given("A first time user lands on the status desktop and generates new key") 
+def step(context):
+    welcomeScreen.agree_terms_conditions_and_generate_new_key()
+
+    
+@When("user inputs username |any| and password |any|") 
+def step(context, username, password):
+    welcomeScreen.input_username_and_password_and_finalize_sign_up(username, password)
+
+    
+@Then("the user lands on the signed in app") 
+def step(context):
+    StatusChatScreen()
```

Comment:
That `Then` implementation doesn't say many information to me. Maybe here it should be called a method like `chatScreenObj.loaded()` to better understand that what we are expecting here is a specific screen loaded.

Reply:
yes I will do that once my app stops crashing